clap_complete/src/dynamic/completer.rs

```diff
@@ -53,6 +53,7 @@ pub fn complete(
     let mut current_cmd = &*cmd;
     let mut pos_index = 1;
     let mut is_escaped = false;
+    let mut _state = ParseState::Unknown;
```

Comment:
Why does `_state` start with a `_`?

Reply:
`_state` is still an unused variable. Starting with `_` is just to prevent warnings.

Reply:
When presented with a warning like that, it would be good to consider carefully before "suppressing" it.  In this case, the compiler was basically telling you that the commit is not atomic.